Content.Server/Botany/Systems/PlantAnalyzerSystem.cs

```diff
@@ -0,0 +1,287 @@
+using Content.Server.Botany.Components;
+using Content.Server.PowerCell;
+using Content.Shared.DoAfter;
+using Content.Shared.Interaction;
+using Content.Shared.PlantAnalyzer;
+using Robust.Server.GameObjects;
+using Robust.Shared.Audio.Systems;
+using Robust.Shared.Player;
+using Robust.Shared.Prototypes;
+using System.Linq;
+
+namespace Content.Server.Botany.Systems;
+
+public sealed class PlantAnalyzerSystem : EntitySystem
+{
+    [Dependency] private readonly PowerCellSystem _cell = default!;
+    [Dependency] private readonly SharedAudioSystem _audio = default!;
+    [Dependency] private readonly SharedDoAfterSystem _doAfterSystem = default!;
+    [Dependency] private readonly UserInterfaceSystem _uiSystem = default!;
+
+    [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+
+    private PlantHolderComponent _plantHolder = default!;
+    private Boolean _scanMode;
```

Comment:
This should not be stored in the system. This appears to duplicate the `AdvancedScan` state variable in the component.